src/lib.rs

```diff
@@ -137,6 +146,17 @@ impl Expr {
         }
     }
 
+    /// Converts this expression to a SqlQuery Struct
+    /// with parameters separated to use with parameter binding
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use cql2::Expr;
+    ///
+    /// let expr = Expr::Bool(true);
+    /// let s = expr.as_sql();
+    /// ```
     pub fn as_sql(&self) -> SqlQuery {
```

Comment:
Separate from this PR, IMO this method should be named `to_sql` because it's doing a non-free conversion (i.e. it allocates a new `Vec`). `as_sql_inner` should probably be `to_sql_inner` as well for the same reason (it uses `to_string`).

Reply:
hah, didn't even know that was the difference in convention between "to_" and "as_" I was just trying to make them internally consistent.

Reply:
Yeah, it's a subtle thing but pretty pervasive so worth nailing into the lib early:

- `as_` for conversion that don't cost (e.g. if a struct stores a string internally and you're just returning a reference to it)
- `to_` for conversions that cost
- `into_` for conversions that consume (e.g. `fn into_foo(self) -> Foo`

Ref: https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv